### ISSUE TITLE
uPIT

### ISSUE THREAD

**Issue by ujscjj** [pchao6/LSTM_PIT_Speech_Separation#5]
你好。我对uPIT有一个盲点，一直搞不清楚，想向你请教一下。uPIT是针对整条语句的，那它是把一条语音作为一个样本送进网络进行训练的吗？ 还是我的理解是错的呢？谢谢你哈。

**Comment by aishoot** [comment]
@ujscjj 是的。之所以这么做是因为如果按照帧进行训练输出，在测试时因为没有先验信息就不知道一个T-F bin属于哪一个说话人。与此同时，实验证明句子级别的置换不变训练)的方法相对于多类回归的模型也更加容易收敛，在没有见过的说话人甚至语言上也有很好的泛化性能。

**Comment by ujscjj** [comment]
谢谢老哥。（弱弱地问一句，我能加一下你的联系方式嘛。0.0      我的QQ：1372179162）

**Comment by ujscjj** [comment]
#

**Comment by ujscjj** [comment]
@pchao6 谢谢你。（弱弱地问一句，我能加一下你的联系方式嘛。0.0      我的QQ：1372179162）

**Comment by JayYang-Fdu** [comment]
学长，你有实现上面的那个danet这个方法嘛，我最近在做那个，参考的是https://github.com/khaotik/DaNet-Tensorflow 这个代码我用的timit数据集，实现不了唉，我看您跟吴学长有沟通过这个，不知道有没有实现呢。
另一个问题就是，我能用timit数据集去实现你的这个代码吗？因为我最近才学的这方面的东西，有很多东西不明白，生成txt那个地方我就不明白。。。。感谢你的回复！

**Comment by ujscjj** [comment]
你好。不好意思，我没有实现过 attractor network 的代码。祝好。




------------------ 原始邮件 ------------------
发件人: "Yangjie55"<notifications@github.com>;
发送时间: 2019年5月16日(星期四) 上午9:38
收件人: "pchao6/LSTM_PIT_Speech_Separation"<LSTM_PIT_Speech_Separation@noreply.github.com>;
抄送: "一棵树"<1372179162@qq.com>;"State change"<state_change@noreply.github.com>;
主题: Re: [pchao6/LSTM_PIT_Speech_Separation] uPIT (#5)




学长，你有实现上面的那个danet这个方法嘛，我最近在做那个，参考的是https://github.com/khaotik/DaNet-Tensorflow 这个代码我用的timit数据集，实现不了唉，我看您跟吴学长有沟通过这个，不知道有没有实现呢。
 另一个问题就是，我能用timit数据集去实现你的这个代码吗？因为我最近才学的这方面的东西，有很多东西不明白，生成txt那个地方我就不明白。。。。感谢你的回复！

**Comment by aishoot** [comment]
@Yangjie55 最近我正在看并实现DANet。TIMIT数据集可以用这个代码的，简单点的话就是把TIMIT文件的组织形式改成WSJ0的文件组织形式，不然就修改代码

**Comment by nagasaibharath** [comment]
I am currently doing this Project. And I have TIMIT dataset with me. So, Can you please explain how to use this code you have written for WSJ0 dataset to be used with TIMIT dataset. You can conact me personally also.(nagasai_bharath@yahoo.in)

**Comment by aishoot** [comment]
@nagasaibharath Hi, you just need to save the mixed audio to folders in the same format as WSJ0's.

**Comment by Rajkashyapmallala** [comment]
Hi,can you explain the architecture of the project? TIA
